helix-core/src/main/java/org/apache/helix/task/RuntimeJobDag.java

```diff
@@ -146,8 +146,13 @@ public boolean finishJob(String job) {
     }
     // Add finished job's successors to ready-list
     if (_isJobQueue) {
-      if (_lastJob != null && _parentsToChildren.containsKey(_lastJob)) {
-        _readyJobList.offer(_parentsToChildren.get(_lastJob).iterator().next());
+      while (_lastJob != null && _parentsToChildren.containsKey(_lastJob)) {
```

Comment:
Better to have some comments here. I think this is an optimization instead of fix. Even if you add the scheduled one as next job. It will be skipped in schedule job part in WorkflowDispatcher. 

Would that cause problem for get another next job?

Reply:
I will add the comments. Actually this is a fix. If you look at the test I added, it is failing without this fix. If you run the newly added test, you will notice that the jobs are not running in parallel because of this issue.

Reply:
+1. I believe we're making this improvement because we've seen an issue in production. Could you add a block comment here what the issue was and why this would fix the issue to give more context?

Reply:
Sure. Added comments about it. I will change the PR for master branch.